helm/calico-capz/templates/capz-calico.yaml

```diff
@@ -0,0 +1,1242 @@
+      - name: calico-node-startup
+        image: sigwindowstools/calico-node:v3.23.0-hostprocess
+        args: ["$env:CONTAINER_SANDBOX_MOUNT_POINT/calico/node-service.ps1"]
+        workingDir: "$env:CONTAINER_SANDBOX_MOUNT_POINT/calico/"
+        imagePullPolicy: Always
+        volumeMounts:
+        - name: calico-config-windows
+          mountPath: /etc/kube-calico-windows/
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: CNI_IPAM_TYPE
+          value: "calico-ipam"
+        - name: CALICO_NETWORKING_BACKEND
+          value: "vxlan"
+        - name: KUBECONFIG
+          value: "C:/etc/cni/net.d/calico-kubeconfig"
+        - name: VXLAN_VNI
+          value: "4096"
+      - name: calico-node-felix
+        image: sigwindowstools/calico-node:v3.23.0-hostprocess
+        args: ["$env:CONTAINER_SANDBOX_MOUNT_POINT/calico/felix-service.ps1"]
+        imagePullPolicy: Always
+        workingDir: "$env:CONTAINER_SANDBOX_MOUNT_POINT/calico/"
+        volumeMounts:
+        - name: calico-config-windows
+          mountPath: /etc/kube-calico-windows/
+        - name: calico-static-rules
+          mountPath: /calico/static-rules.json
+          subPath: static-rules.json
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+        - name: VXLAN_VNI
+          value: "4096"
+        - name: KUBECONFIG
+          value: "C:/etc/cni/net.d/calico-kubeconfig"
+      volumes:
+      - name: calico-config-windows
+        configMap:
+          name: calico-config-windows
+      - name: calico-static-rules
+        configMap:
+          name: calico-static-rules
+      # Used to install CNI.
+      - name: cni-bin-dir
+        hostPath:
+          path: /opt/cni/bin
+      - name: cni-net-dir
+        hostPath:
+          path: /etc/cni/net.d
+      - name: kubeadm-config
+        configMap:
+          name: kubeadm-config
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    k8s-app: kube-proxy
+  name: kube-proxy-windows
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      k8s-app: kube-proxy-windows
+  template:
+    metadata:
+      labels:
+        k8s-app: kube-proxy-windows
+    spec:
+      serviceAccountName: kube-proxy
+      securityContext:
+        windowsOptions:
+          hostProcess: true
+          runAsUserName: "NT AUTHORITY\\system"
+      hostNetwork: true
+      containers:
+      - image: sigwindowstools/kube-proxy:{{ .Capabilities.KubeVersion }}-calico-hostprocess
```

Comment:
@jsturtevant @marosset the existing implementation replaces `+` chars with `_` chars:

`${KUBERNETES_VERSION/+/_}`

Does that suggest that we need to do this string replacement foo in helm as well to ensure that pre-release CI versions work?

Reply:
Yes I believe so.

Reply:
Yes, also need this updated https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/0187833c20b134404ec85b2280e7fd253effaba1/templates/test/ci/prow-ci-version/patches/kubeadm-bootstrap-windows.yaml#L34